### PR TITLE
Add GitHub PR and Issue templates.

### PR DESCRIPTION
**Pull request by hjoliver**
Adds templates for:
- Pull Request, with requirements check list
- Issues:
   - Bug Report
   - Feature Request

@matthewrmshin - please review or reassign.

If approved I'll post the same as a PR to master.

Based on templates in one of @kinow's repositories (which you can also play with to see how they work): https://github.com/kinow/testlink-java-api/tree/master/.github  ... thanks @kinow 👍

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+COMMENT: Choose one of:
+Changes to partially address #xxxx
+Changes to Close #xxxx
+Small change with no associated Issue.
+
+COMMENT: Check all requirements now or later. Mark the PR as a Draft if later.
+- [ ] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
+- [ ] Minimal number of commits with logically grouped changes (use interactive
+         rebase to tidy your branch history if necessary).
+- [ ] No off-topic changes changes (use other PRs for these).
```
Comment (matthewrmshin) [inline .github/PULL_REQUEST_TEMPLATE.md]: Duplicated `changes` here?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+- [ ] Tests with adequate coverage included (unit and/or functional).
+- [ ] Entry added to the change log `CHANGES.md` for next release.
```
Comment (matthewrmshin) [inline .github/PULL_REQUEST_TEMPLATE.md]: The only problem with this is that PRs will likely conflict with one another...

Comment (kinow) [reply .github/PULL_REQUEST_TEMPLATE.md]: That does happen in pandas. Then the developers normally ask contributors to rebase the pull request. e.g. https://github.com/pandas-dev/pandas/pull/25103#pullrequestreview-224761435

A bit annoying for the contributor, but takes 2 minutes. And saves a lot of time for the developers. Guess it depends on whether the pros are worth for us.

Comment (matthewrmshin) [reply .github/PULL_REQUEST_TEMPLATE.md]: (I am happy to re-base. Actually pretty obsessed with re-basing..., but others may not be so happy.)

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (hjoliver) [reply .github/PULL_REQUEST_TEMPLATE.md]: oops

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (hjoliver) [reply .github/PULL_REQUEST_TEMPLATE.md]: I'm a massive rebaser.  Note recommendation to use it tidy your branch history - makes review easier too.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (hjoliver) [reply .github/PULL_REQUEST_TEMPLATE.md]: fixed

**Comment by hjoliver** [comment]
PR Template tidied up a bit, while removing the typo spotted by @matthewrmshin

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (hjoliver) [reply .github/PULL_REQUEST_TEMPLATE.md]: (I'll take this as resolved in favor of adding change log entries).

**Pull request merged**